sql/analyzer/indexed_joins.go

```diff
@@ -282,6 +286,74 @@ func addLookupJoins(m *Memo) error {
 	})
 }
 
+// convertSemiToInnerJoin adds inner join alternatives for semi joins.
+// The inner join plans can be explored further.
```

Comment:
(minor) Something to at least think about... it might be helpful to drop breadcrumb links in docs or even a very short description for readers who aren't familiar with the rules for when to convert X to Y. We can read through the code to reverse engineer the rules, but having them written out would be less error prone and faster for new code readers, and just seems like helpful documentation for the codebase.